java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/buildsteps/docker/DefaultDockerfileGenerator.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.runtimes.builder.buildsteps.docker;
+
+import com.google.cloud.runtimes.builder.injection.JarRuntimeImage;
+import com.google.cloud.runtimes.builder.injection.ServerRuntimeImage;
+import com.google.cloud.runtimes.builder.util.FileUtil;
+import com.google.inject.Inject;
+
+import java.nio.file.Path;
+
+/**
+ * Default implementation of a {@link DockerfileGenerator}.
+ */
+public class DefaultDockerfileGenerator implements DockerfileGenerator {
+
+  private static final String DOCKERFILE = "FROM %s\n"
+      + "ADD %s %s\n";
+
+  private final String jarRuntime;
+  private final String serverRuntime;
+
+  /**
+   * Constructs a new {@link DefaultDockerfileGenerator}.
+   */
+  @Inject
+  DefaultDockerfileGenerator(@JarRuntimeImage String jarRuntime,
+      @ServerRuntimeImage String serverRuntime) {
+    this.jarRuntime = jarRuntime;
+    this.serverRuntime = serverRuntime;
+  }
+
+  @Override
+  public String generateDockerfile(Path artifactToDeploy) {
+    String fileType = FileUtil.getFileExtension(artifactToDeploy);
+    String baseImage;
+    String appDest;
+
+    // TODO both runtimes should suppport adding the application to the same destination
+    if (fileType.equals("jar")) {
+      baseImage = jarRuntime;
+      appDest = "app.jar";
+    } else if (fileType.equals("war")) {
+      baseImage = serverRuntime;
+      appDest = "/app";
```

Comment:
I'm happy to make that change since we're currently not there anyway, but this does move us further away from having all runtimes support applications being added to the same destination. I'll log an issue to track this since it will require some thinking.